docs/src/main/asciidoc/firestore.adoc

```diff
@@ -39,19 +42,147 @@ include::../../test/java/org/springframework/cloud/gcp/sample/firestore/Firestor
+To begin using this library, add the `spring-cloud-gcp-data-firestore` artifact to your project.
+
+Maven coordinates for this module only, using <<getting-started.adoc#_bill_of_materials, Spring Cloud GCP BOM>>:
+
+
+[source,xml]
+----
+<dependency>
+  <groupId>org.springframework.cloud</groupId>
+  <artifactId>spring-cloud-gcp-data-firestore</artifactId>
+</dependency>
+----
+
+Gradle coordinates:
+
+[source]
+----
+dependencies {
+  compile group: 'org.springframework.cloud', name: 'spring-cloud-gcp-data-firestore'
+}
+----
+
+We provide a Spring Boot Starter for Spring Data Firestore, with which you can use our recommended auto-configuration setup. To use the starter, see the coordinates below.
+
+[source,xml]
+----
+<dependency>
+  <groupId>org.springframework.cloud</groupId>
+  <artifactId>spring-cloud-gcp-starter-data-firestore</artifactId>
+</dependency>
+----
+
+Gradle coordinates:
+
+[source]
+----
+dependencies {
+  compile group: 'org.springframework.cloud', name: 'spring-cloud-gcp-starter-data-firestore'
+}
+----
+
+===== Reactive Repository settings
+
+Spring Data Repositories can be configured via the `@EnableReactiveFirestoreRepositories` annotation on your main `@Configuration` class.
+With our Spring Boot Starter for Spring Data Cloud Firestore, `@EnableReactiveFirestoreRepositories` is automatically added.
+It is not required to add it to any other class, unless there is a need to override finer grain configuration parameters provided by https://github.com/spring-cloud/spring-cloud-gcp/blob/master/spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/repository/config/EnableReactiveFirestoreRepositories.java[`@EnableReactiveFirestoreRepositories`].
+
+===== Autoconfiguration
+
+Our Spring Boot autoconfiguration creates the following beans available in the Spring application context:
+
+- an instance of `FirestoreTemplate`
+- instances of all user defined repositories extending `FirestoreReactiveRepository` (an extension of `ReactiveCrudRepository` with additional Cloud Firestore features) when repositories are enabled
+- an instance of https://developers.google.com/resources/api-libraries/documentation/firestore/v1/java/latest/[`Firestore`] from the Google Cloud Java Client for Firestore, for convenience and lower level API access
+
+==== Object Mapping
+
+Spring Data Cloud Firestore allows you to map domain POJOs to https://firebase.google.com/docs/firestore/data-model#collections[Cloud Firestore collections] and documents via annotations:
+
+[source,java,indent=0]
+----
+
+
+include::../../../../spring-cloud-gcp-data-firestore/src/test/java/org/springframework/cloud/gcp/data/firestore/User.java[tag=class_definition]
+
+----
+
+NOTE: Internally we use Firestore client library object mapping. See https://developers.google.com/android/reference/com/google/firebase/firestore/package-summary[the documentation] for supported annotations.
+
+=== Reactive Repositories
```

Comment:
This header looks like a duplicate of `=== Spring Data Reactive Repositories for Cloud Firestore`.
It feels like the sections above should go below this section.